### PR TITLE
[MM-44707] Add CallChannelState type

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

Adds some common types to be used in https://github.com/mattermost/mattermost-plugin-calls/pull/406

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-44707

### CODE REVIEW

File: lib/types/types.d.ts [hunk 1]
```diff
@@ -87,3 +87,24 @@ export type CallsConfig = {
...
 };
+export type CallUserState = {
+    unmuted: boolean;
+    raised_hand: number;
+};
+export type CallState = {
+    id: string;
+    start_at: number;
+    users: string[];
+    states?: CallUserState[];
```
Comment (cpoile) [inline lib/types/types.d.ts:98]: The following is compared to what I have on mobile (just want to double check):
states is not optional (but could be an empty [])

File: lib/types/types.d.ts [hunk 2]
```diff
@@ -87,3 +87,24 @@ export type CallsConfig = {
...
+    thread_id: string;
+    post_id: string;
+    screen_sharing_id: string;
+    owner_id: string;
+    host_id: string;
+    recording?: CallRecordingState;
```
Comment (cpoile) [inline lib/types/types.d.ts:104]: on mobile recording wasn't optional

File: lib/types/types.d.ts [hunk 3]
```diff
@@ -87,3 +87,24 @@ export type CallsConfig = {
...
+};
+export type CallChannelState = {
+    enabled: boolean;
+    channel_id: string;
+    call: CallState;
```
Comment (cpoile) [inline lib/types/types.d.ts:109]: on mobile, enabled and call are optional

**Comment by streamer45** [comment]
@cpoile I think at some point mobile should import these instead as they match what the server is returning.

```go
type RecordingStateClient struct {
	InitAt  int64  `json:"init_at"`
	StartAt int64  `json:"start_at"`
	EndAt   int64  `json:"end_at"`
	Err     string `json:"err,omitempty"`
}

type UserStateClient struct {
	Unmuted    bool  `json:"unmuted"`
	RaisedHand int64 `json:"raised_hand"`
}

type CallStateClient struct {
	ID              string                `json:"id"`
	StartAt         int64                 `json:"start_at"`
	Users           []string              `json:"users"`
	States          []UserStateClient     `json:"states,omitempty"`
	ThreadID        string                `json:"thread_id"`
	PostID          string                `json:"post_id"`
	ScreenSharingID string                `json:"screen_sharing_id"`
	OwnerID         string                `json:"owner_id"`
	HostID          string                `json:"host_id"`
	Recording       *RecordingStateClient `json:"recording,omitempty"`
}

type ChannelStateClient struct {
	ChannelID string           `json:"channel_id,omitempty"`
	Enabled   *bool            `json:"enabled,omitempty"`
	Call      *CallStateClient `json:"call,omitempty"`
}
```

Any field with `omitempty` should be considered as optional.

**Pull request merged**